subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/SkipEmptySourceFilesTaskExecuter.java

```diff
@@ -101,7 +104,7 @@ public TaskExecuterResult execute(TaskInternal task, TaskStateInternal state, fi
                 } else {
                     state.setOutcome(TaskExecutionOutcome.NO_SOURCE);
                 }
-                taskArtifactState.snapshotAfterTaskExecution(context);
+                executionHistoryStore.remove(task.getPath());
```

Comment:
@wolfs I just realized that this should be one line lower, no? We should remove the history even if there were no output files during the previous execution.

Reply:
I guess it's okay to leave it like this on `release`. I've moved it lower on `master`.

Reply:
I guess such a task should always produce outputs when it has sources.

I would only remove the history if there is any, i.e. if `previousExecution != null`.